### ISSUE TITLE
Could not convert RGB mask to yolo format

### ISSUE THREAD

**Issue by Yan-Xuan-4832** [ultralytics/ultralytics#22043]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

I found a RGB masked dataset which I feel good for my project, but convert_segment_masks_to_yolo_seg doesn't work, due to the mismatch of img.shape value number between RGB and gray scale images. I think opencv somehow changed what image.shape work, so gray scale image also have 3 values

### Environment

Ultralytics 8.3.196  Python-3.10.0 torch-2.8.0+cpu CPU (Intel Core i9-10885H 2.40GHz)
Setup complete  (16 CPUs, 31.7 GB RAM, 505.6/952.2 GB disk)

OS                     Windows-10-10.0.26100-SP0
Environment            Windows
Python                 3.10.0
Install                pip
Path                   C:\Users\dev07\PycharmProjects\yolo\.venv\Lib\site-packages\ultralytics
RAM                    31.73 GB
Disk                   505.6/952.2 GB
CPU                    Intel Core i9-10885H 2.40GHz
CPU count              16
GPU                    None
GPU count              None
CUDA                   None

numpy                   2.2.6>=1.23.0
matplotlib              3.10.6>=3.3.0
opencv-python           4.12.0.88>=4.6.0
pillow                  11.3.0>=7.1.2
pyyaml                  6.0.2>=5.3.1
requests                2.32.5>=2.23.0
scipy                   1.15.3>=1.4.1
torch                   2.8.0>=1.8.0
torch                   2.8.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision             0.23.0>=0.9.0
psutil                  7.0.0
polars                  1.33.0
ultralytics-thop        2.0.17>=2.0.0

### Minimal Reproducible Example

from ultralytics.data.converter import convert_segment_masks_to_yolo_seg

convert_segment_masks_to_yolo_seg(masks_dir=r"C:\Users\dev07\PycharmProjects\yolo\data\WHDLD\ImagesPNG", output_dir=r"C:\Users\dev07\PycharmProjects\yolo\data\WHDLD\ImagesMasks",  classes=6)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Yan-Xuan-4832, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will also assist you here shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. In addition to the snippet you shared, please include:
-1–3 small sample mask files (the exact RGB masks causing the issue)
-Exact error message and full traceback output
-A note on mask encoding (RGB colors per class vs. single-channel indexed), file format (e.g., PNG), and any preprocessing steps
-Confirmation of how the images are read (e.g., OpenCV defaults) so we can align assumptions about image shape and channels

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the report. We are aware of this bug and we are working on a fix. We will provide an update when the fix has been merged. Sorry for the inconvenience.

**Comment by UltralyticsAssistant** [comment]
A potential fix is now available in the [merged pull request](https://github.com/ultralytics/ultralytics/pull/21070) by @Laughing-q with contributions from @lmycross, @Y-T-G, @glenn-jocher.

Added a Windows-specific `imread_unicode` patch that supports non-ASCII image paths while preserving native OpenCV grayscale behavior.

Please test the merged change using this repository's documented workflow. If the issue persists, share the updated behavior and any new diagnostic details. Thank you for reporting it! 🙏
